osu.Game/Rulesets/Scoring/DrainingHealthProcessor.cs

```diff
@@ -41,16 +41,29 @@ public partial class DrainingHealthProcessor : HealthProcessor
         /// </summary>
         private const double max_health_target = 0.4;
 
-        private IBeatmap beatmap;
+        /// <summary>
+        /// The drain rate as a proportion of the total health drained per millisecond.
+        /// </summary>
+        public double DrainRate { get; private set; } = 1;
```

Comment:
I know this is coming from the old code, but how about we get rid of this weird default value? It would mean the whole health would drain in 1 millisecond, which makes little sense.

```diff
diff --git a/osu.Game/Rulesets/Scoring/DrainingHealthProcessor.cs b/osu.Game/Rulesets/Scoring/DrainingHealthProcessor.cs
index a8808d08e5..4a651e2650 100644
--- a/osu.Game/Rulesets/Scoring/DrainingHealthProcessor.cs
+++ b/osu.Game/Rulesets/Scoring/DrainingHealthProcessor.cs
@@ -44,7 +44,7 @@ public partial class DrainingHealthProcessor : HealthProcessor
         /// <summary>
         /// The drain rate as a proportion of the total health drained per millisecond.
         /// </summary>
-        public double DrainRate { get; private set; } = 1;
+        public double DrainRate { get; private set; }
 
         /// <summary>
         /// The beatmap.
@@ -139,8 +139,6 @@ protected override void Reset(bool storeResults)
         {
             base.Reset(storeResults);
 
-            DrainRate = 1;
-
             if (storeResults)
                 DrainRate = ComputeDrainRate();
 
```